src/features/auth/useWalletAuth.ts

```diff
@@ -0,0 +1,44 @@
+import { client } from 'lib/gql/client';
+import { atom, selector, useRecoilValue } from 'recoil';
+
+import storage from 'utils/storage';
+
+import { getAuthToken, setAuthToken } from './token';
+
+import { IAuth } from 'types';
+
+const logout = async (): Promise<boolean> => {
+  const { logoutUser } = await client.mutate(
+    { logoutUser: { id: true } },
+    { operationName: 'logout' }
+  );
+  return !!logoutUser?.id;
+};
+
+const updateToken = ({ address, authTokens }: IAuth) => {
+  const token = address && authTokens[address];
+  if (!token && getAuthToken(false)) logout();
```

Comment:
what's the point of calling `getAuthToken(false)` here? Just to see if token on line 19 is unavailable, but the token could still be stored and retrievable from getAuthToken() ?

Reply:
Hmm, I'm carrying over this logic from [before](https://github.com/coordinape/coordinape/pull/1690/files#diff-fe13ed674a354105a9e96354ec08b53830b3a8f3653f90c4eb1311f6a464e2a1L52), but I think it's true that this could just be `if (!token) logout()`, because i can't think of a case where you would be saving a blank token and there isn't already a token stored. Will test removing this

Reply:
Oh, that's right, this is needed when you load the page for the first time and the user's not logged in yet. `updateToken` is called with blank auth info because the `rWalletAuth` atom is being initialized.